File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/commands/attachments/Get.java
package net.ravendb.clientapi.commands.attachments;

import net.ravendb.abstractions.data.Attachment;
import net.ravendb.abstractions.data.AttachmentInformation;
import net.ravendb.abstractions.data.Etag;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;


@SuppressWarnings("deprecation")
public class Get {

  @SuppressWarnings("unused")
  private interface IFoo {
    //region get_1_0
    @Deprecated
    public Attachment getAttachment(String key);
    //endregion

    //region get_2_0
    @Deprecated
    public AttachmentInformation[] getAttachments(int start, Etag startEtag, int pageSize);
    //endregion
  }

  @SuppressWarnings("unused")
  public Get() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      //region get_1_1
      Attachment attachment = store.getDatabaseCommands().getAttachment("albums/holidays/sea.jpg"); // null if does not exist
      //endregion
    }

    try (IDocumentStore store = new DocumentStore()) {
      //region get_2_1
      AttachmentInformation[] attachments = store.getDatabaseCommands().getAttachments(0, Etag.empty(), 10);
      //endregion
    }
  }
}

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/Attachments/Get.cs
using System.IO;
using Raven.Client.Documents;

namespace Raven.Documentation.Samples.ClientApi.Session.Attachments
{
	public class Get
	{
		private interface IFoo
		{
			#region get_1_0
			Attachment GetAttachment(string key);
			#endregion

			#region get_2_0
			AttachmentInformation[] GetAttachments(int start, Etag startEtag, int pageSize);
			#endregion
		}

		public Get()
		{
			using (var store = new DocumentStore())
			{
				#region get_1_1
				Attachment attachment = store
					.DatabaseCommands
					.GetAttachment("albums/holidays/sea.jpg"); // null if does not exist

				Stream data = attachment.Data();
				#endregion
			}

			using (var store = new DocumentStore())
			{
				#region get_2_1
				AttachmentInformation[] attachments = store
					.DatabaseCommands
					.GetAttachments(start: 0, startEtag: Etag.Empty, pageSize: 10);
				#endregion
			}
		}
	}
}
